include/boost/iterator/iterator_archetypes.hpp

```diff
@@ -426,15 +426,16 @@ namespace detail
       >::type iterator_category;
 
       // Needed for some broken libraries (see below)
-      typedef boost::iterator<
-          iterator_category
-        , Value
-        , typename traversal_archetype_base<
+      struct workaround_iterator_base
+      {
+        typedef typename iterator_archetype_base::iterator_category iterator_category;
+        typedef Value value_type;
+        typedef typename traversal_archetype_base<
               Value, AccessCategory, TraversalCategory
-          >::difference_type
-        , typename access::pointer
-        , typename access::reference
-      > workaround_iterator_base;
+          >::difference_type difference_type;
+        typedef typename access::pointer pointer;
+        typedef typename access::reference reference;
+      };
   };
 }
 
```

Comment:
I do not see the reason for this particular change. It does not seem to have anything to do with inheriting from std::iterator or including the deprecated boost/iterator.hpp.

Reply:
OK, I can see where boost::iterator is just std::iterator lifted into the boost namespace and you are replacing code which directly uses std::iterator with types in std::iterator. I do not know whether this falls under the idea of not inheriting from std::iterator in c++17 but the effect is the same so the code looks fine and I lift my previous objection.